core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -272,18 +309,24 @@ class CachedDataFrame(df: Dataset[Row], val queryString: String,
       newpls: mutable.ArrayBuffer[ParamLiteral]): Unit = {
     if (allbcplans.nonEmpty && !firstAccess) {
       allbcplans.foreach { case (bchj, refs) =>
+        println(s"Repreparing for bcplan = ${bchj} with new pls = ${newpls.toSet}")
         val broadcastIndex = refs.indexWhere(_.isInstanceOf[Broadcast[_]])
         val newbchj = bchj.transformAllExpressions {
           case pl@ParamLiteral(_, _, p) =>
             val np = newpls.find(_.pos == p).getOrElse(pl)
-            ParamLiteral(np.value, np.dataType, p)
+            val x = ParamLiteral(np.value, np.dataType, p)
+            x.considerUnequal = true
+            x
         }
+        println(s"newbchj = ${newbchj}")
```

Comment:
Comment out the println as above or replace with logDebug. We should make it a practice to use logDebug's for all such cases which can be enabled selectively for the class when required in log4j.properties